### DISCUSSION TITLE
L2arc is filled during write operations

### DISCUSSION THREAD

**Discussion by ZakSer66** [openzfs/zfs#11796] [Q&A]
I created zpool with cache ssd disks. As I have read l2arc is usedfor caching read operations from zpool. But when I made a bulk copy to my zpool, I checked iostat with zpool iostat -v and discovered that I have write operations to my cache disks. As I am working in video post-production industry , we have a very little proportion of what is shown to what was shot ( e.g. we have 80TB of footage and we will use (read from array) only 2TB. So making a cache during reading of our project driven sofware from array will fill 2TB of L2ARC to accelerate reading. But the thing is that during write operations to array ( 80 TB) L2ARC is also filled ,so our usefull cache is evicted. And also ssd disks are approaching TBW at a very fast rate, and this is quite discouraging. Can anyone explain or describe how to disable writing to L2ARC during write operations to array (zpool) to preserve warmed cache in L2ARC and save ssd resource?

**Comment by bghira** [comment]
have you tuned any values?

**Comment by amotin** [comment]
L2ARC does not work in context of reading or writing. It tries to save blocks that are going to be evicted from ARC soon.  Also consider there are different L2ARC write policies before ARC filled to the max first time and after that.

**Comment by ZakSer66** [comment]
I did tuned nothing. As it was described in  the triangle  with ARC, L2ARC, ZIL and STORAGE area, L2ARC is filling from ARC during reading operations and ZIL is used during writing operations. So is it POSSIBLE to tune parameters to AVOID filling of L2ARC form ARC during write operations? It would be nice to keep warmed cache after massive write of new data to the pool.  
Another possible decision is  to remove cache from the pool, make new copy and return cache back to the pool. Is it possible to retain the content of the cache after these manipulations?

**Comment by IvanVolosyuk** [comment]
Removing the cache will invalidate the content of the cache. Instead you
can temporary change:
$ zfs set secondarycache=none [dataset]

**Reply by ZakSer66** [reply]
Thank you. This can play the game. I will definitely try it

**Comment by gamanakis** [comment]
You can also try setting the zfs parameter l2arc_mfuonly=1. This will cache only the most frequently used data and metadata (ie accessed at least twice).
